### ISSUE TITLE
Prompt downloading " package name" under progress bar 

### ISSUE THREAD

**Issue by Neo-29** [calamares/calamares#1873]
Greetings, 

Is it possible to have a prompt like " downloading (package name) under progress bar as I know  we've got now available ( ex: pacman installing ( package name)) as a new feature. 

Why? In fact if the module package with try remove is executed the last package removed"stays" prompted under the bar during the downloading time of any following package. 
That could be long regarding a slow internet connection and make the user think the " removing  ( package) takes too much time 
That's what I could think as user as I created my own installer. 

Hope I'm clear enough 

Tks

**Comment by Neo-29** [comment]
Exactly..... And If you enable  debug you can have the prompt in the terminal.. But... As far as for a full group ( plasma for example) it doesn't show the details of the downloading packages.. It's the same pb

**Comment by adriaandegroot** [comment]
The infrastructure is there in the *packages* module. `pacman` is the only package-manager right now that does extensive progress reporting. To do extensive progress reporting, the *packages* module needs to be able to read output from the package manager and turn lines of output into progress reports.

I'm marking this as "help: wanted" because it needs someone who actually uses `pacman` to run that program and build the processing into the *packages* module. To @Neo-29 it would already help if you report **what** `pacman` outputs. How can you tell it's downloading something (from the text output on not-a-terminal)? Right now, Calamares looks for a line that says `:: package changes` and then assumes that every line of output after that is "useful" and reports it as progress.

**Comment by Neo-29** [comment]
Hello. Tks for the answer.
In fact  ( I can put screenshots  if needed) the terminal and the bar show the same.
Just the " installing" part
The " downloading part ( so done just before) is yet not shown either in the terminal or in the bar
This is THAT part we need.
We know it's downloading because we chose the packages.
Example:
If we install xfce4 ( group with many packages) it will show in the terminal pacman - S  - - noconfirm xfce4 only 
Nothing during the downloading phase. Neither  in terminal nor in the bar .. AND then installing each package of the group... ( prompt in  terminal and progress bar)

So to sum up just missing the downloading process, of the packages.

Hope I'm clear enough... If you want any precise informations or pictures.. Let me know about it 

Best regards

**Comment by Neo-29** [comment]
Hello.
Tks I'll rebuild and see if any difference
Tks for your reactivity

**Comment by Neo-29** [comment]
Tested.
What is now prompted in the bar gives more information, that's a progress BUT no "downloading ( name of packages) appears.
It's more verbose in the terminal.
When the terminal says retrieving packages ( name already prompted in the terminal just before), the information given in the bar is not the one but the previous operation...
Maybe things to improve.
If needed I'll give pictures. You tell me....

**Comment by dalto8** [comment]
> What is now prompted in the bar gives more information, that's a progress BUT no "downloading ( name of packages) appears.

In my testing, it did display the packages downloading.  However, there are pauses which I suspect is due to pacman's parallel downloads.  Roughly how many packages are you installing?

Do you have a link to an ISO I could test?

**Comment by Neo-29** [comment]
In fact no.. I'm doing it on my own personal installer for testing purpose.
But I faced exactly the same with pauses.. Which made the installer hang ad vita eternam 😉.
I had to reset.
Ii didn't disoalt the downloading phzde each time I tried
I tried with gnome... It did ( strangely not all packages 🙄) but hung
It didn't with lxqt ( I don't know why)

My installer is only netinstall based. So easy to test.

**Comment by dalto8** [comment]
Well...it works here for me so if you can't provide me with a way to reproduce the issue, I can't do much to help.

**Comment by Neo-29** [comment]
So you always have the "downloading packages" in calamares bar? That's what you say?
I'll post pictures tomorrow to show what I'm facing.
It'll surely be clearer.
Tks for your answer

**Comment by dalto8** [comment]
> So you always have the "downloading packages" in calamares bar?

Yes.

> I'll post pictures tomorrow to show what I'm facing.

I don't think a picture is going to help me reproduce it.  I need access in a way that I can run it and see what is happening.

**Comment by Neo-29** [comment]
Nothing more than a net install with netmodules
It's quite close to arcolinuxD if you want a good example to try.

**Comment by dalto8** [comment]
Again, I tested a netinstall and it works for me.

Testing arco won't help unless he has an ISO with that code that also doesn't work.

**Comment by Neo-29** [comment]
I thing he hasn't build the iso with the next version of calamares yet.
But the code of the iso is accessible and calmares config too on arcolinux github.

**Comment by dalto8** [comment]
> But the code of the iso is accessible and calmares config too on arcolinux github.

Feel free build an arco ISO with the latest code that has the issue and post me a link to it.

**Comment by Neo-29** [comment]
I'll have to find time for that.🤔🤔🤔😔 But I'm sure Erik will soon build new isos with the new calamares.
He's very active as you provable know and releases isos each month I think.

I will anyway post some pictures just to let you know what I'm precisely talking about.
Tks a lot again

And as we say in french

Bonne soirée 😊

**Comment by Neo-29** [comment]
Here's what I mean
What do you think?
![image](https://user-images.githubusercontent.com/90937180/150526080-5c067188-90b1-414a-bea0-1f22e2f82ffb.png)

**Comment by dalto8** [comment]
That part is not the downloading part.  That is the list of packages to review.  The downloading part comes after, "Retrieving Packages" and as you can see from the log, there is no output yet.

**Comment by Neo-29** [comment]
I know that. But There's no input of the dowloading part after... 
Nowhere in calamares bar...

**Comment by dalto8** [comment]
> But There's no input of the dowloading part after...
> Nowhere in calamares bar...

If pacman doesn't output what it is downloading, I have no way to display it.



> Not all the Downloading packages are shown
> Just retrieving ( and believe me it's already dowloading..) for quite a long time.. Then some ( few) packages shown dowloading and then installing all

My best guess is this is caused by the parallel downloads feature.  You could try disabling it and seeing if it makes the output come in more smoothly.

**Comment by Neo-29** [comment]
Tks for the advice. Whre can I disable it? In Archiso config or is there a config in calamares for that purpose?

**Comment by dalto8** [comment]
In `/etc/pacman.conf` in the installed system

**Comment by Neo-29** [comment]
Test it right now...

**Comment by Neo-29** [comment]
I can confirm 100% that the only package show " dowloading..." is the last package of a group ( example kde)  not the others.

You can see all of them in the terminal just before  " retrieving packages" but surely not in calamares bar except for the situation I stated above

Maybe there is nothing more to do...

If... Don't bother asking
